tests/tt_metal/tt_metal/test_kernels/dataflow/unit_tests/command_queue/random_program.cpp

```diff
@@ -0,0 +1,43 @@
+
+// SPDX-FileCopyrightText: © 2023 Tenstorrent Inc.
+//
+// SPDX-License-Identifier: Apache-2.0
+
+
+#ifdef DATA_MOVEMENT
+#include <stdint.h>
+#include "dataflow_api.h"
+#endif
+
+#ifdef COMPUTE
+#include <cstdint>
+#include "compute_kernel_api/common.h"
+#include "compute_kernel_api/tile_move_copy.h"
+#include "compute_kernel_api/eltwise_unary/eltwise_unary.h"
+#include "compute_kernel_api/eltwise_unary/sfpu_split_includes.h"
+#endif
+
+#ifdef DATA_MOVEMENT
+namespace {
+void kernel_main() {
+#endif
+#ifdef COMPUTE
+namespace NAMESPACE {
+void MAIN  {
+#endif
+    constexpr volatile uint32_t outer_loop = get_compile_time_arg_val(0);
+    constexpr volatile uint32_t middle_loop = get_compile_time_arg_val(1);
+    constexpr volatile uint32_t inner_loop = get_compile_time_arg_val(2);
+
+    #pragma unroll(get_compile_time_arg_val(0))
```

Comment:
Probably can pass in some other parameter for unrolling

Reply:
should this be in `test_kernels/dataflow` if it can act as compute and data movement?